### PR TITLE
Add Hard Negative Mining Layer.

### PR DESCRIPTION
**Pull request by hertschuh**
This layer extract negative logits and labels.

### CODE REVIEW

**Review by abheesht17 — CHANGES_REQUESTED** [review]
Thanks for the PR! Left some minor comments

File: keras_rs/src/layers/loss/hard_negative_mining_test.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import keras
+from absl.testing import parameterized
+from keras import ops
+
+from keras_rs.src import testing
+from keras_rs.src.layers.loss import hard_negative_mining
+
+
+class HardNegativeMiningTest(testing.TestCase, parameterized.TestCase):
```
[the remark this answers is not in the document]
Comment (hertschuh) [reply keras_rs/src/layers/loss/hard_negative_mining_test.py]: Also added to `FeatureCross`

**Review by abheesht17 — APPROVED** [review]
LGTM, thanks!
